gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterManager.java

```diff
@@ -142,23 +142,25 @@
 
   public GobblinClusterManager(String clusterName, String applicationId, Config sysConfig,
       Optional<Path> appWorkDirOptional) throws Exception {
+    // Set system properties passed in via application config. As an example, Helix uses System#getProperty() for ZK configuration
+    // overrides such as sessionTimeout. In this case, the overrides specified
+    // in the application configuration have to be extracted and set before initializing HelixManager.
+    GobblinClusterUtils.setSystemProperties(sysConfig);
+
+    //Add dynamic config
+    this.config = GobblinClusterUtils.addDynamicConfig(sysConfig);
```

Comment:
Q: Does the location of adding `dynamicConfig` matter? Is dynamicConfig specifically used for SSL-relevant configs?

Reply:
Yes. Dynamic config generation should happen after system properties are set, since some of the config generation (such as temp file locations) depend on system properties.